### PR TITLE
PSR-2, support backend settings, TYPO3 8.4 compat

### PR DESCRIPTION
**Pull request by vertexvaar**
Hi Philipp,

here goes the first PR. As promised this includes a lot of changes as follows:

- [x] reformat according to PSR-2
- [x] fix all PHPMD/PHPCS issues
- [x] set license to GPL-2.0 (from GPL-2.0**+**)
- [x] increase overall readability
- [x] support settings from EXT:backend which are
        * login logo
        * login background image
        * login highlight color
- [ ] include HTOP/TOPT lib via composer
- [ ] have a look at the `TYPO3 loginProvider` setting
- [ ] Give QrFields a treat as well

I know it's a huge PR and will take some time to review, but i promise i'll will make smaller ones from now on ;)

### CODE REVIEW

File: Resources/Private/Language/locallang.xlf [hunk 1]
```diff
@@ -12,6 +12,12 @@
 			<trans-unit id="login.submit">
 				<source>Validate token</source>
 			</trans-unit>
+			<trans-unit id="login.error.message">
+				<source>Failed validating the token</source>
```
Comment (pgampe) [inline Resources/Private/Language/locallang.xlf]: Failed to validate the token

File: Resources/Private/Layouts/Default.html [hunk 2]
```diff
@@ -4,6 +4,17 @@
 			<div class="typo3-login-wrap">
 				<div class="panel panel-lg panel-login">
 					<div class="panel-body tx_authenticator_login_wrap">
+						<div class="typo3-login-logo">
+							<img src="{logo}" class="typo3-login-image" alt=""/>
```
Comment (pgampe) [inline Resources/Private/Layouts/Default.html]: alt="TYPO3 logo"

File: composer.json [hunk 3]
```diff
@@ -15,9 +15,7 @@
 			"role": "Developer"
 		}
 	],
-	"license": [
-		"GPL-2.0+"
-	],
+	"license": "GPL-2.0",
```
Comment (pgampe) [inline composer.json]: Should stay "GPL-2.0+" to indicate that any later version is fine too.

https://spdx.org/spdx-specification-21-web-version#h.twlc0ztnng3b

File: Classes/Hooks/UserAuthHook.php [hunk 4]
```diff
@@ -90,6 +91,29 @@ protected function showForm($token)
...
+            $css .= '.panel-login { border: 0; } ';
+            $css .= '.typo3-login { background-image: url("' . $backgroundImage . '"); }';
+            $css .= '}';
+            $documentTemplate->inDocStylesArray[] = $css;
+        }
+
+        if (!empty($backendExtConf['loginLogo'])) {
+            $logo = $backendExtConf['loginLogo'];
+        } else {
+            if (!empty($backendExtConf['loginHighlightColor'])) {
```
Comment (vertexvaar) [inline Classes/Hooks/UserAuthHook.php]: I just copied that from \TYPO3\CMS\Backend\Controller\LoginController::main to be sure it does the same as the core (TYPO3 8.4)

Comment (vertexvaar) [inline Classes/Hooks/UserAuthHook.php]: Ah, just saw in the notification mail you meat the CSS part. Fixed that.

File: Resources/Private/Layouts/Default.html (hunk 2 above)
Comment (pgampe) [inline Resources/Private/Layouts/Default.html]: ok

**Comment by pgampe** [comment]
I would be ready to merge. Any objections?

Btw: You should always create an new branch before submitting a pull request to easy closing the branch after the pull request is merged. Otherwise each new commit to that branch will be added to the pull request.

**Comment by vertexvaar** [comment]
Ad: License change: 
GPL-2.0+ is deprecated according to spdx.org: https://spdx.org/licenses/GPL-2.0+.html
Moreover i think composer/packagist does not support GPL-2.0+.
I'd suggest a single or dual license [GPL-2.0 and GPL-3.0] if possible, if dual licensing with two GPL versions is actually possible (please don't ask me, i've found nothing on google, neither am i a lawyer ^^)
I'll cut that proposal from the history to prevent further blocking of this PR. It should be a separate discussion, if it's worth the discussion at all ;)

**Comment by pgampe** [comment]
Regarding the license, this is just a strange confusion on the spec site. I use the GPL v2 license with a modifier which then looks like the deprecate license (it is deprecate because it can be replace with the modifier).

**Comment by vertexvaar** [comment]
Okay then ;)
I think were safe now and you can merge.

**Pull request merged**